src/libData/AccountData/Account.cpp

```diff
@@ -126,36 +203,112 @@ bool Account::IncreaseNonce()
+
+    SHA2<HASH_TYPE::HASH_VARIANT_256> sha2;
+    sha2.Update(DataConversion::StringToCharArray(_k));
+    const vector<unsigned char>& k_hash = sha2.Finalize();
+
+    m_storage.insert(dev::h256(k_hash), rlpStream.out());
+
     m_storageRoot = m_storage.root();
 }
 
-vector<string> Account::GetKeys()
+vector<string> Account::GetStorage(string _k) const
+{
+    if (!isContract())
+        return {};
+
+    SHA2<HASH_TYPE::HASH_VARIANT_256> sha2;
+    sha2.Update(DataConversion::StringToCharArray(_k));
+    const vector<unsigned char>& k_hash = sha2.Finalize();
+
+    dev::RLP rlp(m_storage.at(dev::h256(k_hash)));
+    // mutable, type, value
+    return {rlp[1].toString(), rlp[2].toString(), rlp[3].toString()};
+}
+
+Json::Value Account::GetStorageJson() const
 {
-    vector<string> ret;
+    if (!isContract())
+        return Json::arrayValue;
+    Json::Value root;
     for (auto i : m_storage)
     {
-        ret.push_back(i.first.toString());
+        dev::RLP rlp(i.second);
+        string tVname = rlp[0].toString();
+        string tMutable = rlp[1].toString();
+        string tType = rlp[2].toString();
+        string tValue = rlp[3].toString();
+        LOG_GENERAL(INFO,
+                    "\nvname: " << tVname << " \nmutable: " << tMutable
+                                << " \ntype: " << tType
+                                << " \nvalue: " << tValue);
+        if (tMutable == "False")
+        {
+            continue;
+        }
+        Json::Value item;
+        item["vname"] = tVname;
+        item["type"] = tType;
+        if (tType == "Map" || tType == "ADT")
+        {
+            Json::CharReaderBuilder builder;
+            std::unique_ptr<Json::CharReader> reader(builder.newCharReader());
+            Json::Value obj;
+            string errors;
+            if (reader->parse(tValue.c_str(), tValue.c_str() + tValue.size(),
+                              &obj, &errors))
+            {
+                item["value"] = obj;
+            }
+            else
+            {
+                LOG_GENERAL(
+                    WARNING,
+                    "The map json object cannot be extracted from Storage: "
+                        << errors);
+            }
+        }
+        else
+        {
+            item["value"] = tValue;
+        }
+        root.append(item);
     }
-    return ret;
-}
+    Json::Value _balance;
+    _balance["vname"] = "_balance";
+    _balance["type"] = "Int";
+    int balance = static_cast<int>(GetBalance());
+    _balance["value"] = to_string(balance);
+    root.append(_balance);
 
-vector<string> Account::GetStorage(string _k)
-{
-    dev::RLP rlp(m_storage[_k]);
-    return {rlp[0].toString(), rlp[1].toString(), rlp[2].toString()};
-}
+    LOG_GENERAL(INFO, "States: " << root);
 
-string Account::GetStorageValue(string _k) { return GetStorage(_k)[2]; }
+    return root;
+}
 
 void Account::RollBack()
 {
+    if (!isContract())
```

Comment:
enclosed in bracket and maybe log warning if neccessary

Reply:
Noted